web/client/components/TOC/fragments/LayerMetadataModal.jsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2015, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const React = require('react');
+const PropTypes = require('prop-types');
+const {Glyphicon} = require('react-bootstrap');
+
+require("./css/settingsModal.css");
+
+const Dialog = require('../../misc/Dialog');
+const Portal = require('../../misc/Portal');
+const Template = require('../../data/template/jsx/Template');
+const MetadataTemplate = require('./template/MetadataTemplate');
+
+class LayerMetadataModal extends React.Component {
+    static propTypes = {
+        id: PropTypes.string,
+        layerMetadata: PropTypes.object,
+        hideLayerMetadata: PropTypes.func,
+        closeGlyph: PropTypes.string,
+        panelStyle: PropTypes.object,
+        panelClassName: PropTypes.string,
+        maskLoading: PropTypes.bool,
+        layerMetadataPanelTitle: PropTypes.string
+    };
+
+    static defaultProps = {
+        id: "mapstore-layer-settings-metadata",
+        layerMetadata: {expanded: false},
+        closeGlyph: "1-close",
+        panelStyle: {},
+        panelClassName: "toolbar-panel portal-dialog",
+        maskLoading: true,
+        layerMetadataPanelTitle: ''
+    };
+
+    onDelete = () => {
+        this.props.hideLayerMetadata();
+    };
+
+    onClose = () => {
+        this.props.hideLayerMetadata();
+    };
+
+    renderBodyTemplate = () => {
+        if (!this.props.layerMetadata.maskLoading) {
+            return (
+                <Template
+                    model={this.props.layerMetadata.metadataRecord}
+                    template={MetadataTemplate} />
```

Comment:
I would add support for react-bootstrap components in the template. Have a look at renderContent property of the template.
There is an example here:

https://github.com/geosolutions-it/csi-sira/blob/master/frontend/js/components/template/TemplateSira.jsx
https://github.com/geosolutions-it/csi-sira/blob/master/frontend/js/components/template/index.js